op-e2e/actions/proofs/sequence_window_expiry_test.go

```diff
@@ -48,6 +48,82 @@ func runSequenceWindowExpireTest(gt *testing.T, testCfg *helpers.TestCfg[any]) {
 	env.RunFaultProofProgram(t, l2SafeHead.Number.Uint64()/2, testCfg.CheckResult, testCfg.InputParams...)
 }
 
+// Runs a that proves a block in a chain where the batcher opens a channel, the sequence window expires, and then the
+// batcher attempts to close the channel afterwards.
+func runSequenceWindowExpire_ChannelCloseAfterWindowExpiry_Test(gt *testing.T, testCfg *helpers.TestCfg[any]) {
+	t := actionsHelpers.NewDefaultTesting(gt)
+	tp := helpers.NewTestParams()
+	env := helpers.NewL2FaultProofEnv(t, testCfg, tp, helpers.NewBatcherCfg())
+
+	// Mine 2 empty blocks on L2.
+	for i := 0; i < 2; i++ {
+		env.Sequencer.ActL2StartBlock(t)
+		env.Sequencer.ActL2EndBlock(t)
+	}
+
+	// Open the channel on L1.
+	env.Batcher.ActL2BatchBuffer(t)
+	env.Batcher.ActL2BatchSubmit(t)
+	env.Miner.ActL1StartBlock(12)(t)
+	env.Miner.ActL1IncludeTxByHash(env.Batcher.LastSubmitted.Hash())(t)
+	env.Miner.ActL1EndBlock(t)
+
+	// Finalize the block with the first channel frame on L1.
+	env.Miner.ActL1SafeNext(t)
+	env.Miner.ActL1FinalizeNext(t)
+
+	// Instruct the sequencer to derive the L2 chain from the data on L1 that the batcher just posted.
+	env.Sequencer.ActL1HeadSignal(t)
+	env.Sequencer.ActL2PipelineFull(t)
+
+	// Ensure the safe head is still 0.
+	l2SafeHead := env.Engine.L2Chain().CurrentSafeBlock()
+	require.EqualValues(t, 0, l2SafeHead.Number.Uint64())
+
+	// Cache the next frame data before expiring the sequence window, but don't submit it yet.
+	env.Batcher.ActL2BatchBuffer(t)
+	env.Batcher.ActL2ChannelClose(t)
+	finalFrame := env.Batcher.ReadNextOutputFrame(t)
+
+	// Expire the sequence window by building `SequenceWindow + 1` empty blocks on L1.
+	for i := 0; i < int(tp.SequencerWindowSize)+1; i++ {
+		env.Alice.L1.ActResetTxOpts(t)
+		env.Alice.ActDeposit(t)
+
+		env.Miner.ActL1StartBlock(12)(t)
+		env.Miner.ActL1IncludeTx(env.Alice.Address())(t)
+		env.Miner.ActL1EndBlock(t)
+
+		env.Miner.ActL1SafeNext(t)
+		env.Miner.ActL1FinalizeNext(t)
+	}
+
+	// Instruct the batcher to closethe channel on L1, after the sequence window + channel timeout has elapsed.
+	env.Batcher.ActL2BatchSubmitRaw(t, finalFrame)
+	env.Miner.ActL1StartBlock(12)(t)
+	env.Miner.ActL1IncludeTxByHash(env.Batcher.LastSubmitted.Hash())(t)
+	env.Miner.ActL1EndBlock(t)
+
+	// Finalize the block with the second channel frame on L1.
+	env.Miner.ActL1SafeNext(t)
+	env.Miner.ActL1FinalizeNext(t)
```

Comment:
Why do we need to finalise the L1 blocks?

Reply:
We don't necessarily need to, but figured it's worthwhile to ensure that the full L1 head state is updated (across all labels.) We have the flexibility to not do this, tho.

I think it'd be nice to re-work the API a bit to hide this - Normally, I figure that this is what we want to do when advancing the head, but might as well save the granular `ActL1<Label>Next` calls for when we need to only advance certain labels.

Reply:
I was assuming we wouldn't normally do this and test that op-node is advancing based on the latest L1 blocks. It shouldn't be waiting for them to become safe or finalized.